src/gov/nasa/worldwind/symbology/milstd2525/MilStd2525TacticalSymbol.java

```diff
@@ -540,15 +540,18 @@ protected void layoutDynamicModifiers(DrawContext dc, AVList modifiers, OrderedS
             double length = this.iconRect.getHeight();
             Boolean directionOnly = true;
             Object d = this.getModifier(SymbologyConstants.SPEED_LEADER_SCALE);
-            if (d != null && d instanceof Number)
+            if (d != null && d instanceof Number) {
                 directionOnly = false;
                 length *= ((Number) d).doubleValue();
-
+            }
             if (this.useGroundHeadingIndicator)
             {
                 List<? extends Point2D> points = MilStd2525Util.computeGroundHeadingIndicatorPoints(dc, osym.placePoint,
-                    (Angle) o, length, this.iconRect.getHeight(), directionOnly);
-                this.addLine(dc, Offset.BOTTOM_CENTER, points, LAYOUT_RELATIVE, points.size() - 1, osym);
+                        (Angle) o, length, this.iconRect.getHeight(), directionOnly);
+                if (directionOnly)
+                    this.addLine(dc, Offset.BOTTOM_CENTER, points, LAYOUT_RELATIVE, points.size() - 4, osym);
+                else
+                    this.addLine(dc, Offset.BOTTOM_CENTER, points, LAYOUT_RELATIVE, points.size() - 1, osym);
```

Comment:
When useGroundHeadingIndicator is true and if addLine method gets points.size()-1 as argument, when directionOnly is true, then the anchor point for the MIlitary Symbol calculated wrong and when the earth is rotated Military Symbol does not show proper position on the earth at certain view headings.